⌨️ feat: Improve keyboard navigation and shortcuts for Feedback Task

leiyre opened the pull request:
<!-- Thanks for your contribution! As part of our Community Growers initiative 🌱, we're donating Justdiggit bunds in your name to reforest sub-Saharan Africa. To claim your Community Growers certificate, please contact David Berenstein in our Slack community or fill in this form https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

Improve keyboard navigation for Feedback Task:

- [x] Make status dropdown focusable and selectable using keyboard
- [x] Make labels and rating items focusable and selectable
- [x] Simplify code for search label focus status
- [x] Show focused first option in first question in form
- [x] Navigate between  labels using `TAB` 
- [x] Select label using `SPACE`
- [x] Click button using `ENTER`
- [x] Allow navigate between questions using `Shift + ↑` and `Shift + ↓`. 
- [x] Display elements in ranking slots using `1`, `2`, `3`, `4`, `5`, `6`, `7`,`8`,`9`,`10`,`11`, `12` (12 max)

Closes #3264

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**Checklist**

- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

damianpumar commented on frontend/components/commons/forms/questions-form/QuestionsForm.component.vue:
```diff
@@ -40,35 +50,42 @@
+          :isFocused="checkIfQuestionIsFocused(index)"
           :description="input.description"
           v-model="input.options"
+          :key="JSON.stringify(input.options)"
```
Is it necessary the options parsed as string for the shortcuts things?

leiyre replied:
We need to force re-rendering the component after ranking using keyboard, I tried other approach without success, do you have any idea?

leiyre replied:
I have included more logic in children and seems to work

damianpumar commented on frontend/components/commons/forms/questions-form/fields/ranking/Ranking.component.vue:
```diff
@@ -57,6 +66,16 @@ export default {
         }))
       );
     },
+    onChangedWithKeyboard(value, rank) {
```
What is the difference to change the options with the keyboard or the mouse here? I mean, the business logic is the same when the user change the options with keyboard or mouse

leiyre replied:
yes, we can use the same

leiyre committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

* develop:
  refactor: add `HuggingFaceDatasetMixIn` under `integrations` (#3326)
  feat: add list user workspaces endpoint (#3308)
  ci: Stop linking issues to team work project
  chore: add missing `greenlet` dependency in `server` extra (#3330)
  fix: unit test failing if not local db (#3307)
  ci: Optimize build + test pipeline (#3300)
  refactor: simplify old bulk endpoints to avoid create datasets if does not exists (#3306)
  📝 Update doc site link (#3299)
  🚑 Fix dependencies (#3302)
  feat: migrate to async SQLAlchemy engine (#3162)

leiyre committed:
simplify logic in question autofocus data

leiyre committed:
fix styles for focus

leiyre committed:
fix textarea focus when is markdown

leiyre committed:
use same ranking event with keyboard

leiyre committed:
fix lint

leiyre committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

* develop:
  fix: return all workspaces in system for owner users (#3343)
  fix: `rg.init` with argilla user using quickstart images raise an unexpected error (#3341)
  feat: add `Suggestion` endpoints (#3304)
  feat: add `list_user_workspaces` and `User.workspaces` property (#3334)
  docs: update developer docs (#3314)
  Docs/3312 docs 112 is not building correctly (#3313)

leiyre committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

* develop:
  fix: `coroutine` not awaited (#3344)

leiyre committed:
feat: feedback task helpbox (#3305)

<!-- Thanks for your contribution! As part of our Community Growers
initiative 🌱, we're donating Justdiggit bunds in your name to reforest
sub-Saharan Africa. To claim your Community Growers certificate, please
contact David Berenstein in our Slack community or fill in this form
https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

Add shortcuts section to the help box

Closes #2438

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)
- [ ] Documentation update


**Checklist**

- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK)
(see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

---------

Co-authored-by: nataliaElv <natalia@argilla.io>
Co-authored-by: Damián Pumar <damianpumar@gmail.com>

leiyre committed:
remove force re-render

leiyre committed:
Merge branch 'feat/improve_keyboard_navigation' of github.com:argilla-io/argilla into feat/improve_keyboard_navigation

* 'feat/improve_keyboard_navigation' of github.com:argilla-io/argilla:
  feat: feedback task helpbox (#3305)

damianpumar commented on frontend/components/commons/forms/questions-form/fields/shared-components/drag-and-drop-selection/DndSelection.component.vue:
```diff
@@ -56,11 +64,90 @@ export default {
+    },
   },
   methods: {
+    rankWithKeyboard(event, item) {
```
I would like us to review this method together

leiyre committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

* develop:
  feat: disable concurrency tests GitHub (#3355)
  feat: improve protected fields (#3342)
  chore: added telemetry to ArgillaTrainer (#3336)
  Feat/update start page code (#3327)
  [DOCS] Added information about tests in developer documentation (#3147)

# Conflicts:
#	docs/_source/_common/snippets/start_page.md
#	frontend/components/commons/datasets-list/DatasetsEmpty.vue

leiyre committed:
fix enter keypress error on sumbit

leiyre committed:
unit test rankWithKeyboard

damianpumar committed:
♻️ Refactor rankWithKeyboard

leiyre committed:
include icons and change font weight

Commit:
docs: update shortcuts table

leiyre committed:
update submit shortcut

leiyre committed:
improve autofocus visualization

leiyre committed:
fix style

leiyre committed:
prevent submit

leiyre committed:
allow expand labels on tab

leiyre committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

* develop:
  ci: skip running tests when no python changes (#3357)
  [pre-commit.ci] pre-commit autoupdate (#3323)
  feat: warn users when using integer record ids (#3328)
  fix: resolved import errors `ArgillaTrainer` (#3346)
  fix: Set pydantic < 2.0 (#3349)
  docs: add Dolly video tutorial (#3335)

leiyre committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

* develop:
  fix: `ArgillaDatasetCard` to handle all question `options` (#3366)
  🎉 Add user workspaces into user setting page (#3353)
  fix: `buildChanges` output had the value of `pythonChanges` (#3363)

leiyre committed:
include has-suggestion directive

leiyre committed:
fix const name

leiyre committed:
include vuex model

leiyre committed:
include suggestion model and queries

leiyre committed:
fetch and store suggestions

leiyre committed:
get suggestions in component

damianpumar committed:
Merge branch 'develop' into feat/frontend_suggestions

leiyre committed:
wip

damianpumar committed:
🎉 Improve feedback task

leiyre committed:
wip

damianpumar committed:
💄 Suggestion working, WIP

damianpumar committed:
🚀 Improve

leiyre committed:
wip

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

damianpumar committed:
💄 Fix code

damianpumar committed:
Merge branch 'feat/frontend_suggestions' of https://github.com/argilla-io/argilla into feat/frontend_suggestions

damianpumar committed:
🎉 Remove debugger

damianpumar committed:
🚀 Fix

damianpumar committed:
💄 Improve code

damianpumar committed:
✨ Add suggestion for ranking component

damianpumar committed:
💄 Deprecate function

leiyre committed:
show emoji with tooltip when question has suggestion

leiyre committed:
update check is question has suggestion

leiyre committed:
remove no-unused-vars

damianpumar committed:
💄 Improve code

damianpumar committed:
Merge branch 'feat/frontend_suggestions' of https://github.com/argilla-io/argilla into feat/frontend_suggestions

damianpumar committed:
📝 Improve has suggestion method

damianpumar committed:
🚀 Refactor some classes

Co-authored-by: leiyre <leiyre@users.noreply.github.com>
Co-authored-by: Keith Cuniah <keithCuniah@users.noreply.github.com>

damianpumar committed:
🚀 Fix minor issue

leiyre committed:
rename directive

damianpumar committed:
💄 Fix

damianpumar committed:
🚑 Revert!

keithCuniah committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

leiyre committed:
fix hasSuggestion name

leiyre committed:
remove console.log

leiyre committed:
fix: prevent show tooltip on hovering text

keithCuniah committed:
:sparkles: renaming

keithCuniah committed:
:sparkles: renaming and css

keithCuniah committed:
:sparkles: add `shortcut help` in panel && renaming

keithCuniah committed:
:coffin: remove obsolete object

keithCuniah committed:
:recycle: renaming && removing obsolete code

damianpumar committed:
💄 Working

leiyre committed:
create new sidebar group for shortcuts

leiyre committed:
include shortcuts icon

leiyre committed:
update sidebar tooltip

leiyre committed:
update shortcuts button group, type and styles

leiyre committed:
move styles from iconButton

keithCuniah commented on frontend/components/commons/sidebar-feedback-task/IconButton.vue:
```diff
@@ -103,6 +103,13 @@ export default {
+    border-radius: $border-radius;
+    max-height: 20px;
+    max-width: 20px;
+  }
```
move logic/css for the sidebar outside of the icon component (or the icon component become less generic)

leiyre committed:
explicit import icons

leiyre committed:
styles for table

leiyre committed:
Resolve conflicts

leiyre committed:
fix transition glitch

leiyre committed:
remove tabs and include title

leiyre committed:
duplicate component without tabs

damianpumar committed:
✨ Feedback working, WIP metrics

damianpumar committed:
🚀 Wip Metrics

damianpumar committed:
🚀 Improve code

damianpumar committed:
⚰️ Remove dead code

damianpumar committed:
🚀 Improve, WIP change array offset from entity

damianpumar committed:
🚀  WIP delete unused code and reuse from settings, WIP replace or create new VM

damianpumar committed:
✨ Reuse code

damianpumar committed:
✨ Fix tests

damianpumar committed:
💄 Remove unused class

damianpumar committed:
✨ Improve code and DI

damianpumar committed:
🚑 Update version

damianpumar committed:
✨ Remove unused

damianpumar committed:
✨ Fix beforeUnmount

damianpumar committed:
🚑 Improve loading ux

damianpumar committed:
🐛 Fix

damianpumar committed:
📝 Improve code

damianpumar committed:
🚑 Improve and reuse code

damianpumar committed:
🔥 Improve code

damianpumar committed:
Merge branch 'develop' into fix/tests-and-dependencies

keithCuniah committed:
:sparkles: trap focus on labels for `rating`, `ranking`, `multilabel`, `singlelabel`

Co-authored-by: Damián Pumar <damianpumar@gmail.com>

keithCuniah committed:
Merge branch 'feat/improve_keyboard_navigation' of https://github.com/argilla-io/argilla into feat/improve_keyboard_navigation

keithCuniah committed:
:bug: custom `tabIndex` to not focus with tab

damianpumar committed:
💄 Add babel-core

damianpumar committed:
✨ Fix names

damianpumar committed:
🚀 Register dependencies just once

damianpumar committed:
✨ Remove unnecessary get version info.

keithCuniah committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

keithCuniah committed:
:recycle: compute props to get ref input

keithCuniah committed:
:recycle: refacto big condition

keithCuniah committed:
:sparkles: focus on search on key event

damianpumar committed:
✨ Change folder structure, remove moment

damianpumar committed:
✨ Fix lint

damianpumar committed:
🐛 Fix tests

damianpumar committed:
🚑 Fix

damianpumar committed:
✨ Improve tests

damianpumar committed:
🚑 Lint

damianpumar committed:
📝 Fix test

damianpumar committed:
📝 Fix

damianpumar committed:
✨ Fix tests

damianpumar committed:
🚑 Fix date

damianpumar committed:
Merge branch 'fix/tests-and-dependencies' into feat/improve_keyboard_navigation

damianpumar committed:
🚑 Merged file

damianpumar committed:
✨ Fix

damianpumar committed:
✨ Fix

keithCuniah committed:
:bug: click on active element on press space

keithCuniah committed:
:sparkles: shortcuts on `rating` component

keithCuniah committed:
:coffin: remove not used ref

damianpumar committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

damianpumar committed:
✨ Fix help shortcuts documentation panel

damianpumar committed:
💄 Register directive

keithCuniah committed:
:bug: exit function if no search component

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

keithCuniah committed:
:fire: remove `shift` shortcut for `rating`

keithCuniah committed:
:recycle: ensure `key` is from `Numpad`

damianpumar committed:
🐛 Fix dots in documentation viewer (#3435)

damianpumar committed:
Merge branch 'develop' into refactor/improve-domain-design

damianpumar committed:
🐛 Add e2e tests

keithCuniah committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

keithCuniah committed:
:construction: WIP : debounce to manage more than one digit in shortcut

keithCuniah committed:
:recycle: pass callback to debounce function

damianpumar committed:
💄 Remove unused

damianpumar committed:
🚀 match suggestions

keithCuniah committed:
:sparkles: support shortcut from `Numpad` and `Digit`

damianpumar committed:
💄 Restore record before to return it

damianpumar committed:
💄 Restore record before to return it

damianpumar committed:
🎉 Fix match suggestion

damianpumar committed:
Merge branch 'refactor/improve-domain-design' of https://github.com/argilla-io/argilla into refactor/improve-domain-design

damianpumar committed:
✨ Remove unnecessary code

keithCuniah committed:
:white_check_mark: unit test for shortcut component

damianpumar committed:
🚀 Fix e2e

keithCuniah committed:
:sparkles: select next questions after answering `rating` question

Co-authored-by: Damián Pumar <damianpumar@gmail.com>

damianpumar committed:
Merge branch 'refactor/improve-domain-design' into feat/improve_keyboard_navigation

keithCuniah committed:
:white_check_mark: add missing dependency

keithCuniah committed:
Merge branch 'feat/improve_keyboard_navigation' of https://github.com/argilla-io/argilla into feat/improve_keyboard_navigation

keithCuniah committed:
:white_check_mark: add slot mock

keithCuniah committed:
:construction: skip failling test due to slot implementation

keithCuniah committed:
:bug: add missing properties which break shortcuts

keithCuniah committed:
:sparkles: emit only when user have answer rating

damianpumar committed:
✨ Add some E2E tests related to annotation page

keithCuniah committed:
:fire: remove code for debounce && shortcut `0` to target value === `10` for rating component

keithCuniah committed:
:recycle: add computed

keithCuniah committed:
:sparkles: focus on next question

keithCuniah committed:
:sparkles: focus on next question

keithCuniah committed:
:sparkles: logic for focus outline

damianpumar committed:
🚑 FIx

damianpumar committed:
Merge branch 'refactor/improve-domain-design' into feat/improve_keyboard_navigation

damianpumar committed:
🚑  New test for shortcuts

damianpumar committed:
🎉 Add pictures for shortcuts tests

damianpumar committed:
✨ Fix issue

damianpumar committed:
🐛 Fix bug

damianpumar committed:
🚀  Improve screenshots folder

damianpumar committed:
Merge branch 'refactor/improve-domain-design' into feat/improve_keyboard_navigation

damianpumar committed:
💄 Add missing files

damianpumar committed:
🔥 Fix code

damianpumar committed:
🚀 Move component to correct folder

damianpumar committed:
🚀 Remove unnecessary code

damianpumar committed:
🚀 Fix issue in markdown and discard action with keyboard

damianpumar committed:
🐛 Fix issue in copy button

damianpumar committed:
📝 Remove debugger keyword

damianpumar committed:
🔥 Fix issue

damianpumar committed:
✨ Add e2e for other browsers

damianpumar committed:
Merge branch 'refactor/improve-domain-design' into feat/improve_keyboard_navigation

damianpumar committed:
💄 User can move to every question for all record status

damianpumar committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

Commit:
docs: shortcuts update

damianpumar committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

damianpumar committed:
Merge branch 'feat/improve_keyboard_navigation' of https://github.com/argilla-io/argilla into feat/improve_keyboard_navigation

keithCuniah committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

keithCuniah committed:
:sparkles: FEAT : shortcuts improvements (#3473)

# Description

Closes : [#119](https://github.com/argilla-io/Team/issues/119)

Merge after #3268 

- Ranking question: shortcut for value `10/11/12` is 10/11/12
- Add backspace as shortcut to remove options from ranking
- Form actions (Clear, Discard and Submit) should only have shift+space,
shift+backspace and shift+enter shortcuts
- As for ranking, the rating shortcut for value `10` is 10

**Type of change**

- New feature (non-breaking change which adds functionality)

---------

Co-authored-by: leire <leire@recogn.ai>
Co-authored-by: Damián Pumar <damianpumar@gmail.com>
Co-authored-by: nataliaElv <natalia@argilla.io>

damianpumar committed:
Revert ":recycle: renaming && remove unecessary fn argument"

This reverts commit 5befb467cf183b56147f0930beac308cef003317.

damianpumar committed:
Revert ":recycle: replace formId by formRef && centralize in computed"

This reverts commit ac36f279c4d8398edf29d0720d53de2c64f399c4.

damianpumar committed:
Revert ":white_check_mark: e2e test"

This reverts commit 834bb058fb8f99a9785ab1b0b1939ce1c226ba4d.

damianpumar committed:
Revert ":sparkles: focus on first question when target is in form but not in a question"

This reverts commit b439e5e69966b13ac307d0f6fc83be7c15855235.

damianpumar committed:
✨ Move focus to first position when user comes from outside

damianpumar committed:
🔥 Remove debugger

keithCuniah committed:
:recycle: renaming && replace fn param by computed props

damianpumar committed:
🎉 Update pixel max ratio

keithCuniah committed:
:construction: e2e test

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

damianpumar committed:
✨ Fix images

damianpumar committed:
🔥 New e2e

damianpumar committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

gabrielmbmb committed:
fix: Slack PR message not found

gabrielmbmb committed:
fix: `thread_ts` is not always present

damianpumar committed:
🐛 Fix issue

keithCuniah committed:
:sparkles: always use `shift` for global shortcuts

keithCuniah committed:
:white_check_mark: e2e tests

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

damianpumar committed:
🚀 New e2e tests

damianpumar committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

damianpumar committed:
🐛 Fix issue (#3577)

Fix issue https://github.com/argilla-io/argilla/issues/3576

damianpumar committed:
✨ Improve UX for rating component

damianpumar committed:
🔥 Update screenshots

damianpumar committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

damianpumar committed:
🐛 Fix merge problem

leiyre committed:
style remove shadow

leiyre committed:
style help shortcut height

leiyre committed:
make record transition more explicit

leiyre committed:
fix paste in Textarea

damianpumar committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

damianpumar committed:
🐛 Fix

damianpumar committed:
🔥 Update e2e for user settings page

damianpumar committed:
🔥 Update e2e pictures for datasets page

damianpumar committed:
🔥 Update dataset settings e2e tests

damianpumar committed:
🔥 Update e2e tests for annotation page

damianpumar committed:
⚡️ Fix e2e for annotation page in ranking component

damianpumar committed:
✨ Replace all selectors for keyboard moves

leiyre committed:
fix style for sidebar shortcuts panel

keithCuniah committed:
:sparkles: FEAT : update global shortcuts (#3630)

Update global shortcuts : 
- activate form with `tab` only instead of `shift+arrow`
- go to next / prev record with `ctrl+arrowRight`  `ctrl+arrowLeft`
- documentation panel : the shortcut to go to next/prev records will be
on bottom of the list

Refacto :
- centralized shortcuts in e2e 

**Type of change**

- New feature

**How Has This Been Tested**

- Only feedback task

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Damián Pumar <damianpumar@gmail.com>
Co-authored-by: nataliaElv <natalia@argilla.io>

damianpumar committed:
✨ Upgrade e2e pictures

damianpumar committed:
Merge branch 'develop' into feat/improve_keyboard_navigation

damianpumar committed:
✨ Update e2e

damianpumar committed:
💄 Add new behavior for text component

damianpumar committed:
✨ Edit mode with enter

leiyre committed:
update color for focused label outline

Commit:
📝 update shortcuts table

leiyre committed:
update color for textarea outline

leiyre committed:
Merge branch 'feat/improve_keyboard_navigation' of github.com:argilla-io/argilla into feat/improve_keyboard_navigation

* 'feat/improve_keyboard_navigation' of github.com:argilla-io/argilla:
  📝 update shortcuts table

damianpumar committed:
✨ Fix when no have markdown config

damianpumar committed:
💫 Update shortcuts panel

damianpumar committed:
💫 Update e2e images for autosave

damianpumar committed:
💫 Ranking component e2e images updated

damianpumar committed:
💫 New images for text component

damianpumar committed:
💫 Shortcut panel

damianpumar committed:
💫 Rating component

damianpumar committed:
⚡️ Improve Rating component test

damianpumar committed:
💫 Global shortcuts for text component

damianpumar committed:
✨ Update focus color in annotation page

damianpumar committed:
🐛 Fix issue for Safari

damianpumar committed:
⚡️ Improve selection text

damianpumar committed:
⚡️ Update images for firefox and webkit issues

damianpumar committed:
✨ Wait 1 second to hide the save

damianpumar committed:
🎉 Fix e2e images
